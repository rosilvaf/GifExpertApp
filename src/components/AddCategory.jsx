import { useState } from "react"
import propTypes from 'prop-types'


export default function AddCategory( {onNewCategory}) {

   const [inputValue, setInputValue] = useState('')

   const onInputChange = ({target}) =>{
    setInputValue(target.value)
   }

   const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
    setInputValue('')
    onNewCategory(inputValue.trim())
   }

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input type="text"
        placeholder="Search Gifs"
        value = {inputValue}
        onChange={onInputChange} />
    </form>
    )
}


AddCategory.proptypes = {
    onNewCategory: propTypes.func.isrequired
}