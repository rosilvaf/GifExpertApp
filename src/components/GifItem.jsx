import propTypes from 'prop-types'

//To specify a CSS class, use the className attribute.
export const GifItem = ({title, url, id}) => {
  return (
    <div className="card">
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
  )

}
//PropTypes exporta un rango de validadores que pueden ser usados para estar seguros que la información recibida sea válida. 
GifItem.propTypes = {
  title:propTypes.string.isRequired,
  url: propTypes.string.isRequired
}
