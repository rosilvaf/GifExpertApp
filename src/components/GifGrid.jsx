import GifItem from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGIfs";
import propTypes from 'prop-types';


 
export default function GifGrid({category}) {
    const {images, isLoading} = useFetchGifs(category);
    
    return (
        <>
        <h3>
            {category}
        </h3>
        {
            isLoading && ( <h2>  Loading...   </h2>)
    
        }
        <div className="card-grid">
            {
                images.map(({image}) => (
                   <GifItem key={image.id}
                   {...image}
                   />
                ))
            }
        </div>
        </>
    )
}


GifGrid.prototype = {
    category: propTypes.string.isRequired,   
}