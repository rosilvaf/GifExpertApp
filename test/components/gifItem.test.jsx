import {render, screen} from '@testing-library/react';
import {GifItem} from '../../src/components/GifItem'

describe('Test in <GifItem/>', () =>{
    const title = "hi";
    const url = "https://one-punch.com/hi.jpg";
    test('should do match with the snapshot', ()=>{
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('it should show image with url and alt adecuated', ()=>{
        render( <GifItem title={title} url={url}/>)
        // screen.debug();;
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('it should show the title in the component', ()=>{
        render( <GifItem title={title} url={url}/>)
       
        expect(screen.getByText(title)).toBeTruthy()
    })
})


