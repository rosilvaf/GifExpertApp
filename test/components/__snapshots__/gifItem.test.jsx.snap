// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> should do match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hi"
      src="https://one-punch.com/hi.jpg"
    />
    <p>
      hi
    </p>
  </div>
</div>
`;
